src/test/java/com/aws/greengrass/componentmanager/builtins/S3DownloaderTest.java

```diff
@@ -63,6 +70,7 @@ class S3DownloaderTest {
     private static final String VALID_ARTIFACT_CONTENT = "Sample artifact content";
     private static final String TEST_COMPONENT_NAME = "ComponentWithS3Artifacts";
     private static final String TEST_COMPONENT_VERSION = "1.0.0";
+    private final ExecutorService executor = Executors.newSingleThreadExecutor();
```

Comment:
why do you need this?

Reply:
This looks unused though, so let's not add it